### PR TITLE
Make CRDs installation and ServiceAccount name configurable (#1)

### PR DESCRIPTION
**Pull request by dobrerazvan**
| Q               | A
| --------------- | ---
| Bug fix?        | no
| New feature?    | no
| API breaks?     | no
| Deprecations?   | no
| Related tickets | fixes #X, partially #Y, mentioned in #Z
| License         | Apache 2.0

### What's in this PR?
We needed to have a way to choose if we want to install kafka operator CRDs or not and to name the service account differently. We also started the work to upgrade the chart to helm3 by moving the crds in the crds dedicated folder. This chart maintains compatibility with helm2.

### Why?
It adds more flexibility during deployment where the k8s environment is a restricted and cluster level resources are created by the cluster admin and not the namespace admin. 

### Checklist

- [x] Implementation tested
- [ ] Error handling code meets the [guideline](https://github.com/banzaicloud/pipeline/blob/master/docs/error-handling-guide.md)
- [ ] Logging code meets the guideline
- [ ] User guide and development docs updated (if needed)

### CODE REVIEW

File: charts/kafka-operator/values.yaml [hunk 1]
```diff
@@ -48,13 +51,20 @@ prometheusMetrics:
...
+
 
 nameOverride: ""
 fullnameOverride: ""
 
 rbac:
   enabled: true
 
+crd:
+  enabled: false
```
Comment (amuraru) [inline charts/kafka-operator/values.yaml:66]: let's make sure this is not breaking with previous version. should this be `true` ?

Comment (dobrerazvan) [reply charts/kafka-operator/values.yaml:66]: fair, yeah, by default should be true, fixing

**Comment by amuraru** [comment]
@dobrerazvan the CRDs should also define 
```
annotations:
    "helm.sh/hook": crd-install
```
so they are both helm3 and helm2 compatible (see https://geeksocket.in/posts/helm-2-3-migration/)

**Comment by dobrerazvan** [comment]
@amuraru Fixed the crds by adding that annotation.

**Comment by amuraru** [comment]
@dobrerazvan i split this in #404 and #405 to ease the review.
Can you please close this and move the discussion in the new PRs?

**Pull request closed without merging**